python/paddle/tensor/manipulation.py

```diff
@@ -6749,3 +6749,108 @@ def select_scatter(x, values, axis, index, name=None):
         )
 
         return output
+
+
+def slice_scatter(x, value, axis=0, start=None, stop=None, step=1, name=None):
+    """
+    Embeds the value tensor into x at the given axis. Returns a new tensor instead of a view.
+
+    Args:
+        x (Tensor) : The input Tensor. Supported data types are `bool`, `float16`, `float32`, `float64`, `uint8`, `int8`, `int16`, `int32`, `int64`, `bfloat16`, `complex64`, `complex128`.
+        value (Tensor) : The tensor to embed into x. Supported data types are `bool`, `float16`, `float32`, `float64`, `uint8`, `int8`, `int16`, `int32`, `int64`, `bfloat16`, `complex64`, `complex128`.
+        axis (int) : the dimension to insert the value. Default is 0.
+        start (int, optional) : the start index of where to insert. Default is `None` which will be converted to `0`.
+        stop (int, optional) : the stop index of where to insert. Default is `None` which will be converted to `x.shape[axis]`.
+        step (int, optional) : the step for each insert. Default is 1.
```

Comment:
Because operator of `set_value`can support writing data according to multiple `axes`, `starts`, `ends` and `steps`, it is better that the parameters of `axis`, `start`, `stop`, `step` here can support `int` and `list of int` ？ making this API more user-friendly and powerful

Reply:
If modified according to the above suggestions, the parameter naming suggestion are `axes`, `starts`, `ends` and `strides`, which is consistent with API of `paddle.slice` and `paddle.strided_slice`

Reply:
No ~~~

`list of int` of `axis`, `start`, `stop` and `step` for `slice_scatter` NOT make sense ~ 

For instance:

`x` is a tensor of shape `[8, 8]`, `value` is `[8, 2]`, then the `axis` can ONLY be `1`. If `axis=0` , the values of `value` in axis 1 of length 2 can not be fitted into `x` of length 8. In other words, `list of int` can only be useful of `x` and `value` the same shape, which is not that useful ~~~ BTW, torch not support `list of int` either ~~~

Reply:
From a functional perspective, the functionality of this API should be dual to `paddle.strided_slice`. The former is for writing data, while the latter is for reading data, so supporting `list of int` is more natural, and the underlying operator of `set_value` also supports it. As for the example you gave, if the shape of `value` cannot be broadcasted to the shape after the slice, an error should be reported (by `set_value`). Additionally, it's not just torch's design that is reasonable, but how to make it better for users is what we pursue

Reply:
> From a functional perspective, the functionality of this API should be dual to `paddle.strided_slice`. Because the former is for writing data, while the latter is for reading data, so supporting `list of int` is more natural, and the underlying operator of `set_value` also supports it. As for the example you gave, if the shape of `value` cannot be broadcasted to the shape after the slice, an error should be reported (by `set_value`). 

Not agree ~ 

Especially treat `slice_scatter` as a dual to `paddle.strided_slice`.

There should be a powerful API `set_value` or `set_value_with_tensor`, not `slice_scatter` ~

We should not bother users to map `slice_scatter` from torch, mindspore or other framework to ours, and not mess the API up with more options just because we could support it ~

> Additionally, not all of design of torch are reasonable, but how to make it better for users is what we pursue

Agree and always !!!

Reply:
My opinion is that regardless of whether API name is `set_value` or `slice_scatter`, our goal is to have an API that is functionally dual to `paddle.strided_slice`. Here, this PR is just a subset of this feature, and we don't need to create two APIs with the same parameters. The only difference is that one supports `int` and the other supports `list of int`